### PR TITLE
Improve and update network-ng.md document

### PR DESCRIPTION
**Pull request by imobachgs**
Merge after #28 

It updates and improves the `network-ng.md` document.

* Update current status.
* Add a plan.
* Add information about the UI layer and interface builders.

### CODE REVIEW

File: doc/network-ng.md [hunk 1]
```diff
@@ -38,31 +40,77 @@ other classes.
...
     src/lib/y2network/sysconfig
     ├── config_reader.rb <- READER
     ├── config_writer.rb <- WRITER
-    ├── connection_config_reader_handlers
-    │   ├── eth.rb
-    │   ├── wlan.rb
-    │   └── ...
     ├── connection_config_reader.rb
+    ├── connection_config_readers
+    │   ├── eth.rb
```
Comment (jreidinger) [inline doc/network-ng.md]: just curios is it eth or according to new types ethernet? just that we should be consistent.

File: doc/network-ng.md [hunk 2]
```diff
@@ -38,31 +40,77 @@ other classes.
...
-## Accessing the Configuration
+Last but not least, there are additional classes like {Y2Network::Sysconfig::RoutesFile} and
+{Y2Network::Sysconfig::InterfaceFile} which abstract the details of reading/writing `ifcfg` files.
 
-The `Yast::Lan` module is still the entry point to read and write the network configuration. Basically, it keeps two configuration objects, one for the running system and another want for the wanted configuration.
+#### AutoYaST
+
+AutoYaST is a special case in the sense that it reads the information from a profile, instead of
+using the running system as reference. Additionally, it does not implement a writer because the
+configuration will be written using a different backend (like sysconfig).
```
Comment (jreidinger) [inline doc/network-ng.md]: cannot we have writter for clonning and so writting profile?

File: doc/network-ng.md [hunk 3]
```diff
@@ -101,3 +170,9 @@ reader and will write it to the final system using the one corresponding to the
 | USB             |      |       |
 | Dummy           |      |       |
 | s390 types      |      |       |
+
+## (Short Term) Plan
+
+- [ ] Finish reading/writing interfaces
+- [ ] Move missing UI logic to interface configuration builders
+- [ ] Replace LanItems with the new data model
```
Comment (jreidinger) [inline doc/network-ng.md]: maybe also use that new type class everywhere. So everything use same class to represent types. I think now we need in short term focus on unifying to existing new solutions like that new type class.

**Comment by imobachgs** [comment]
Included in #875.

**Pull request closed without merging**